src/dynamodb-mcp-server/awslabs/dynamodb_mcp_server/server.py

```diff
@@ -266,6 +286,175 @@ async def source_db_analyzer(
         return f'Analysis failed: {str(e)}'
 
 
+@app.tool()
+@handle_exceptions
+async def execute_dynamodb_command(
+    command: str = Field(description="AWS CLI DynamoDB command (must start with 'aws dynamodb')"),
+    endpoint_url: Optional[str] = Field(default=None, description='DynamoDB endpoint URL'),
+    ctx: Optional[Context] = Field(default=None, description='Execution context'),
+):
+    """Execute AWSCLI DynamoDB commands.
+
+    Args:
+        command: AWS CLI command string (e.g., "aws dynamodb query --table-name MyTable")
+        endpoint_url: DynamoDB endpoint URL
+        ctx: Execution context
+
+    Returns:
+        AWS CLI command execution results or error response
+    """
+    # Validate command starts with 'aws dynamodb'
+    if not command.strip().startswith('aws dynamodb'):
+        raise ValueError("Command must start with 'aws dynamodb'")
+
+    # Configure environment with fake AWS credentials if endpoint_url is present
+    if endpoint_url:
+        os.environ['AWS_ACCESS_KEY_ID'] = 'FakeAccessKeyID'  # pragma: allowlist secret
+        os.environ['AWS_SECRET_ACCESS_KEY'] = 'FakeSecretAccessKey'  # pragma: allowlist secret
```

Comment:
These are example values:
* https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html
* https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html
```suggestion
        os.environ['AWS_ACCESS_KEY_ID'] = 'AKIAIOSFODNN7EXAMPLE'  # pragma: allowlist secret
        os.environ['AWS_SECRET_ACCESS_KEY'] = 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'  # pragma: allowlist secret
```